src/routing_table/network_tests.rs

```diff
@@ -17,297 +17,244 @@
+                None => SeededRng::new(),
+            },
+            nodes: HashMap::new(),
+        }
+    }
 
-// impl Network {
-//     /// Creates a new empty network with a seeded random number generator.
-//     fn new(optional_seed: Option<[u32; 4]>) -> Network {
-//         Network {
-//             rng: match optional_seed {
-//                 Some(seed) => SeededRng::from_seed(seed),
-//                 None => SeededRng::new(),
-//             },
-//             nodes: HashMap::new(),
-//         }
-//     }
+    /// Adds a new node to the network and makes it join its new group, splitting if necessary.
+    fn add_node(&mut self) {
+        let name = self.random_free_name(); // The new node's name.
+        if self.nodes.is_empty() {
+            // If this is the first node, just add it and return.
+            assert!(self.nodes.insert(name, RoutingTable::new(name, GROUP_SIZE)).is_none());
+            return;
+        }
 
-//     /// Adds a new node to the network and makes it join its new group, splitting if necessary.
-//     fn add_node(&mut self) {
-//         let name = self.random_free_name(); // The new node's name.
-//         if self.nodes.is_empty() {
-//             // If this is the first node, just add it and return.
-//             assert!(self.nodes.insert(name, RoutingTable::new(name, GROUP_SIZE)).is_none());
-//             return;
-//         }
-//         // Find any node that is close to the new one.
-//         let close_peer = self.close_node(name);
-//         // The new node needs to have exactly the same contacts:
-//         let connecting_peers: Vec<u64> = self.known_nodes(close_peer);
-//         let mut new_table = RoutingTable::new(name, GROUP_SIZE);
-//         for peer in connecting_peers {
-//             assert!(unwrap!(self.nodes.get_mut(&peer)).add(name).is_some());
-//             assert!(new_table.add(peer).is_some());
-//         }
-//         assert!(self.nodes.insert(name, new_table).is_none());
-//         // If there is any node containing group can split now, do so.
-//         while let Some(node) = self.nodes
-//             .iter()
-//             .find(|&(_, table)| match table.should_split() {
-//                 Some(_) => true,
-//                 None => false,
-//             })
-//             .map(|(&node, _)| node) {
-//             self.split(node);
-//         }
-//     }
+        let mut new_table = RoutingTable::new(name, GROUP_SIZE);
+        let mut split_prefix = BTreeSet::new();
+        // TODO: needs to verify how to broadcasting such info
+        for node in self.nodes.values_mut() {
+            match node.add(name) {
+                Ok(result) => {
+                    split_prefix.insert(result);
+                }
+                Err(e) => trace!("failed to add node with error {:?}", e),
+            }
+            match new_table.add(*node.our_name()) {
+                Ok(Some(prefix)) => {
+                    let _ = new_table.split(prefix);
```

Comment:
In line 62, there is TODO: needs to verify how to broadcasting such info, which including this part to be re-considered as well.